recipes-app/node-red-gpio/files/0006-node-red-gpio-add-input-for-DIN-node.patch

```diff
@@ -0,0 +1,48 @@
+From 42cfd1e45afe8a092161540e365799fe83e26d57 Mon Sep 17 00:00:00 2001
+From: Ivan Mikhaylov <ivan.mikhaylov@siemens.com>
+Date: Tue, 31 May 2022 18:19:04 +0000
+Subject: [PATCH] node-red-gpio: add input for DIN node
+
+Add improvement for DIN node which adds a possibility to get input
+connection with different nodes, as example, inject node which may
+control when DIN node reports about current status.
```

Comment:
Is this pattern found elsewhere as well? On first sight, it does not look very intuitive to me.

Reply:
It's common pattern for node-red, as I see from their guides how they trigger functions/other things via timer interval or on startup. Plus of this thing, that DIN shouldn't care about timers or intervals or even about startup checkbuttons, inject node care about such things, anyways, even without connected input node it should work as was before.

Reply:
Looking at https://github.com/node-red/node-red-nodes/blob/master/hardware/BBB/145-BBB-hardware.js#L228, it still uses that on-startup pattern, and I'm not seeing any inputs for DIN nodes. Is that node broken as well?

Reply:
No, it is ok. As I mentioned in [for node-re](https://github.com/siemens/meta-iot2050/pull/323#discussion_r888875710) , in that port I didn't add that send which should send it further, it only triggers on GPIO behavior change which doesn't happen on start.